runtime/onert/core/src/compiler/StaticShapeInference.cc

```diff
@@ -924,6 +924,46 @@ void StaticShapeInferer::visit(const ir::operation::Softmax &op)
   handleSimpleUnaryOp(op, op.getInputs().at(ir::operation::Softmax::Input::INPUT));
 }
 
+void StaticShapeInferer::visit(const ir::operation::SpaceToBatchND &op)
+{
+  const auto output_index = op.getOutputs().at(0);
+  const auto input_idx{op.getInputs().at(ir::operation::SpaceToBatchND::Input::INPUT)};
+  const auto block_shape_idx{op.getInputs().at(ir::operation::SpaceToBatchND::Input::BLOCK_SIZE)};
+  const auto padding_idx{op.getInputs().at(ir::operation::SpaceToBatchND::Input::PADDINGS)};
+
+  ir::Operand &output = _operands.at(output_index);
+  const auto &input = _operands.at(input_idx);
+  const auto &block_shape = _operands.at(block_shape_idx);
+  const auto &padding = _operands.at(padding_idx);
+
+  if (input.info().isDynamic() || block_shape.info().isDynamic() || padding.info().isDynamic())
+  {
+    output.info().setDynamic();
+    _return_has_dynamic_tensor = true;
+    return;
+  }
+
+  // Whether input is constant or not does not affect whether output is dynamic or not
+  if (!(block_shape.isConstant() && padding.isConstant())) // TODO : Check this
```

Comment:
@hyunsik-yoon Actually I refer to the example of other layers. But I am not sure about this meaning, Does it mean block_shape(This is 2nd INPUT) and padding(3nd INPUT) must be constant to do static shape inference, otherwise output is decided during runtime?

Reply:
Resolved by offline conversation : yes.